plugins/org.eclipse.xtext.generator/src/org/eclipse/xtext/generator/parser/antlr/AntlrGrammarComparator.xtend

```diff
@@ -0,0 +1,165 @@
+/*******************************************************************************
+ * Copyright (c) 2015 itemis AG (http://www.itemis.eu) and others.
+ * All rights reserved. This program and the accompanying materials
+ * are made available under the terms of the Eclipse Public License v1.0
+ * which accompanies this distribution, and is available at
+ * http://www.eclipse.org/legal/epl-v10.html
+ *******************************************************************************/
+package org.eclipse.xtext.generator.parser.antlr
+
+import java.util.regex.Matcher
+import java.util.regex.Pattern
+import org.eclipse.xtend.lib.annotations.Accessors
+
+/**
+ * Compares two charSequences of ANTLR grammars token by token.
+ * Ignores differences in white space, and counts line breaks for usable error diagnosis. 
+ * 
+ * @author Christian Schneider - Initial contribution and API
+ * @since 2.9
+ */
+class AntlrGrammarComparator {
+
+	public abstract static class AbstractErrorHandler {
```

Comment:
These information are only tracked for the sake of offering helpful error messages, the comparison logic does not rely on them.